### PR TITLE
Add missing metadata to dimensions api

### LINKED ISSUE #1296: fabric-dimensions-v1 is missing metadata
Not sure how I missed it when I opened #1232, but fabric-dimensions-v1 is also missing metadata. In particular, it's missing a name, icon, author, homepage, issues link, sources link, and a description.

### PR DESCRIPTION
**Pull request by actuallyasmartname**
Closes #1296 

Adds missing fields name, icon, authors, and description.
Looks like this was forgotten in #1233

### CODE REVIEW

File: fabric-dimensions-v1/src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -1,13 +1,25 @@
 {
   "schemaVersion": 1,
   "id": "fabric-dimensions-v1",
+  "name": "Fabric Dimensions API (v1)",
   "version": "${version}",
   "license": "Apache-2.0",
+  "icon": "assets/icon.png",
```
Comment (Technici4n) [inline fabric-dimensions-v1/src/main/resources/fabric.mod.json]: That looks wrong. I think you should copy/paste the icon from another module into `resources/assets/fabric-dimensions-v1/icon.png` and then reference it as `assets/fabric-dimensions-v1/icon.png`.

**Review by Player3324 — APPROVED** [review]
Good, thanks!

**Review by Technici4n — CHANGES_REQUESTED** [review]
Looks good besides the icon.

File: fabric-dimensions-v1/src/main/resources/fabric.mod.json (hunk 1 above)
Comment (actuallyasmartname) [reply fabric-dimensions-v1/src/main/resources/fabric.mod.json]: Resolved in https://github.com/FabricMC/fabric/pull/1428/commits/c1d2ff1c4f7e60e9dc4c7626dc482ff14c4e2dcf

**Review by liach — APPROVED** [review]

**Review by Technici4n — APPROVED** [review]
Good now, thanks for the contribution!

**Pull request merged**